controllers/spacebindingcleanup/spacebinding_cleanup_controller.go

```diff
@@ -94,8 +96,71 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 
 func (r *Reconciler) deleteSpaceBinding(logger logr.Logger, spaceBinding *toolchainv1alpha1.SpaceBinding) error {
 	logger.Info("deleting the SpaceBinding")
+
+	// check if spaceBinding was created from SpaceBindingRequest,
+	// in that case we must delete the SBR and will take care of deleting the SpaceBinding as well
+	spaceBindingReqeuestAssociated := checkSpaceBindingRequestAssociated(spaceBinding)
+	if spaceBindingReqeuestAssociated.found {
+		return r.deleteSpaceBindingRequest(logger, spaceBindingReqeuestAssociated)
+	}
+
+	// otherwise delete the SpaceBinding resource directly ...
+	return r.DeleteSpaceBinding(spaceBinding)
+}
+
+func (r *Reconciler) DeleteSpaceBinding(spaceBinding *toolchainv1alpha1.SpaceBinding) error {
```

Comment:
it's a minor thing but I guess this function doesn't need to be exported so we could name it `deleteSpaceBindingDirectly` or something so that we don't export the function when it doesn't need to be. This wouldn't impact much other than potentially unit tests I guess.